### PR TITLE
prove mhp0cl, mhpinvcl

### PR DESCRIPTION
**Pull request by icecream17**
also:
move fnexd to main

refactor suppco out from supp0cosupp0 and imacosupp

prove suppcofnd

### CODE REVIEW

File: set.mm [hunk 1]
```diff
@@ -74569,10 +74578,50 @@ any sets (which usually are functions) and any element (even not
     $}
   $}
 
+  $( The support of the composition of two sets.  Originally a subproof of
+     ~ supp0cosupp0 .  (Contributed by SN, 15-Sep-2023.) $)
```
Comment (benjub) [inline set.mm:74582]: What I generally write in these cases is
```
  $( The support of the composition of two functions is the inverse image by
     the outer function of the support of the inner function.  (Contributed by
     AV, 30-May-2019.)  Extract this statement from the proof of
     ~ supp0cosupp0 .  (Revised by SN, 15-Sep-2023.) $)
```

File: set.mm [hunk 2]
```diff
@@ -256700,6 +256779,43 @@ Additional definitions for (multivariate) polynomials
...
+    $d g h j $.  $d I g h $.  $d N g $.  $d I f x $.  $d R f x $.  $d X x $.
+    $d ph x $.
+    mhpinvcl.h $e |- H = ( I mHomP R ) $.
+    mhpinvcl.p $e |- P = ( I mPoly R ) $.
+    mhpinvcl.m $e |- M = ( invg ` P ) $.
+    mhpinvcl.i $e |- ( ph -> I e. V ) $.
+    mhpinvcl.r $e |- ( ph -> R e. Grp ) $.
+    mhpinvcl.n $e |- ( ph -> N e. NN0 ) $.
+    mhpinvcl.x $e |- ( ph -> X e. ( H ` N ) ) $.
+    $( Negation is closed for homogeneous polynomials.  (Contributed by SN,
```
Comment (benjub) [inline set.mm:256792]: > $( Homogeneous polynomials are closed under negation.

(or "closed under taking the opposite")

**Review by tirix — APPROVED** [review]

File: set.mm (hunk 1 above)
Comment (benjub) [reply set.mm]: @icecream17 : sorry, I got the "inner" and "outer" mixed up !

**Comment by jkingdon** [comment]
I guess @icecream17 needs to resolve the conflicts but if that happens maybe we can merge this before #3490

**Pull request merged**